assembly/ScaleString.ts

```diff
@@ -0,0 +1,42 @@
+import { Bytes } from "./utils/Bytes"
+
+export class ScaleString extends Array<u8> {
```

Comment:
Maybe the ScaleString can extend the `ByteArray`. This way we can skip the lines `8` to `11` and the `toU8a` String implementation.
If we rename the `toString` in the ByteArray to `toHexString` we might even get one additional feature to the `SCALEString`